src/model/sample.rs

```diff
@@ -324,11 +323,21 @@ impl Sample {
                     let end_pos = cigar.end_pos() as u32;
 
                     if pos <= start && end_pos >= start {
-                        let cigar = record.cigar();
-                        observations.push(
-                            self.read_observation(&record, &cigar, start, variant, chrom_seq)?
-                        );
-                        n_overlap += 1;
+                        if let Some( (prob_ref, prob_alt) ) = evidence::reads::prob_snv(record, &cigar, start, variant, chrom_seq)? {
```

Comment:
Any reason to not put this into read_observations again? It shares a lot of the code...